src/storage/assembler.rs

```diff
@@ -184,8 +193,9 @@ impl Assembler {
 
     /// Add a new contiguous range to the assembler, and return `Ok(())`,
     /// or return `Err(())` if too many discontiguities are already recorded.
-    pub fn add(&mut self, mut offset: usize, mut size: usize) -> Result<(), TooManyHolesError> {
+    pub fn add(&mut self, mut offset: usize, mut size: usize) -> Result<bool, TooManyHolesError> {
```

Comment:
Doc comment should be updated to reflect signature change (the `return Ok(())` bit)

Reply:
Updated now, thanks!